### PR TITLE
#15064 refactor NavToolTest to create test objects

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java [hunk 1]
```diff
@@ -65,134 +71,88 @@ public static void restoreProperty() {
...
+            final int currentShowOnMenuItems = findShowOnMenuUnderFolder(folder, user);
 
-        int spanishResultChildren = navResult.getChildren().size();
-        //We are expecting 2 children result for Spanish Language.
-        assertEquals(spanishResultChildren, 2);
+            //Comparing what we found vs the result on the NavTool
+            int englishResultChildren = navResult.getChildren().size();
 
-        List<IHTMLPage> liveHTMLPages = APILocator.getHTMLPageAssetAPI().getLiveHTMLPages(aboutUsFolder, user, false);
+            assertEquals(englishResultChildren, currentShowOnMenuItems);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java]: change order of the comparison

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java [hunk 2]
```diff
@@ -65,134 +71,88 @@ public static void restoreProperty() {
...
-        NavResult navResult = new NavTool().getNav(demoHost, aboutUsIdentifier.getPath(), 1, user);
-        assertNotNull(navResult);
+            //Create 2 Folders (One with show on Menu and one without)
+            final Folder subFolder1 = new FolderDataGen().folder(folder).showOnMenu(true).nextPersisted();
+            final Folder subFolder2 = new FolderDataGen().folder(folder).showOnMenu(false).nextPersisted();
 
-        //Find out how many show on menu items we currently have
-        final int currentShowOnMenuItems = findShowOnMenuUnderFolderRecursive(aboutUsFolder, user);
+            //Using Identifier to get the path.
+            Identifier aboutUsIdentifier = APILocator.getIdentifierAPI().find(folder);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java]: rename var

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java [hunk 3]
```diff
@@ -65,134 +71,88 @@ public static void restoreProperty() {
...
-        //Now We are expecting same children result for Spanish Language and English Language.
-        assertEquals(englishResultChildren, navResult.getChildren().size());
+            //List of contentlets created for this test.
+            List<Contentlet> contentletsCreated = new ArrayList<>();
 
-        //Now remove all the pages that we created for this tests.
-        APILocator.getContentletAPI().unpublish(contentletsCreated, user, false);
-        APILocator.getContentletAPI().archive(contentletsCreated, user, false);
-        APILocator.getContentletAPI().delete(contentletsCreated, user, false);
+            createSpanishPagesCopy(user, liveHTMLPages, contentletsCreated);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java]: Let's do something like this instead to help clarity:
```List<Contentlet> contentletsCreated = createSpanishPagesCopy(user, liveHTMLPages);```

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java [hunk 4]
```diff
@@ -298,26 +238,45 @@ public void test_getNavWithoutParameters() throws Exception {
...
+        try {
+            //New template
+            final Template template = new TemplateDataGen().nextPersisted();
 
-        Mockito.when(request.getRequestURI()).thenReturn("/about-us");
-        Mockito.when(request.getServerName()).thenReturn("demo.dotcms.com");
-        Mockito.when(viewContext.getRequest()).thenReturn(request);
+            //Create 2 Pages (One with show on Menu and one without)
+            final HTMLPageAsset pageAsset1 = new HTMLPageDataGen(folder, template).showOnMenu(true).languageId(1).nextPersisted();
+            final HTMLPageAsset pageAsset2 = new HTMLPageDataGen(folder, template).showOnMenu(false).languageId(1).nextPersisted();
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java]: we need to publish these two pages:
```
final HTMLPageAsset pageAsset1 = new HTMLPageDataGen(folder, template).showOnMenu(true).languageId(1).nextPersisted();
            pageAsset1.setIndexPolicy(IndexPolicy.FORCE);
            APILocator.getContentletAPI().publish(pageAsset1, user, false);

            final HTMLPageAsset pageAsset2 = new HTMLPageDataGen(folder, template).showOnMenu(false).languageId(1).nextPersisted();
            pageAsset2.setIndexPolicy(IndexPolicy.FORCE);
            APILocator.getContentletAPI().publish(pageAsset2, user, false);

```

**Review by dsilvam — APPROVED** [review]

**Pull request merged**